### ISSUE TITLE
HTTP requests don't redirect to HTTPS for a site set to use HTTPS

### ISSUE THREAD

**Issue by terry-alphasoftware** [alphasoftware/AlphaAnywhere#1]
It is common for a site set to use HTTPS to allow a request to come in over HTTP and then redirect that request to use HTTPS. Add an "Enforce secure connection" option to the security settings to add this behavior so that no post publish configuration on IIS is needed.

**Comment by Sparticuz** [comment]
This is the code I use elsewhere:

```
<system.webServer>
  <rewrite>
    <rules>
      <rule name="Redirect to HTTPS" stopProcessing="true">
        <match url="(.*)" />
        <conditions><add input="{HTTPS}" pattern="^OFF$" /></conditions>
        <action type="Redirect" url="https://{HTTP_HOST}/{R:1}" redirectType="Permanent" />
      </rule>
    </rules>
  </rewrite>
</system.webServer>
```

or in the GUI

![image](https://cloud.githubusercontent.com/assets/161913/10084753/27448ace-62d3-11e5-9ed6-373a09d9349f.png)
![image](https://cloud.githubusercontent.com/assets/161913/10084759/2fedaea8-62d3-11e5-853f-f55d37429c9b.png)

**Comment by terry-alphasoftware** [comment]
@Sparticuz Yes this is typically how you'd do this. The purpose of this issue is to have a publish profile option so that the site can be published from Alpha Anywhere and no other configuration is needed on IIS. I will update my original comment to clarify.
